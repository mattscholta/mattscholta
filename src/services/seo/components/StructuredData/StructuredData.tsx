import * as React from 'react';

interface StructuredDataProps {
  data: any;
}

/**
 * @name StructuredData
 * @external https://developers.google.com/search/docs/guides/intro-structured-data
 * @description Adds JavaScript notation embedded in a <script> tag for
 * SEO (JSON-LD format)
 */
const StructuredData: React.FC<StructuredDataProps> = (props) => {
  const { data } = props;

  // Setup
  const dataString = JSON.stringify(data);

  return (
    <script
      dangerouslySetInnerHTML={{ __html: dataString }}
      type="application/ld+json"
    />
  );
};

export { StructuredData };
export type { StructuredDataProps };
