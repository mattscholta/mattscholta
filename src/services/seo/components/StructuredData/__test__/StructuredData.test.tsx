import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { StructuredData, StructuredDataProps } from '../StructuredData';

describe('StructuredData', () => {
  let props: StructuredDataProps;
  let wrapper: RenderResult;

  const createWrapper = () => <StructuredData {...props} />;

  beforeEach(() => {
    props = { data: {} };
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
