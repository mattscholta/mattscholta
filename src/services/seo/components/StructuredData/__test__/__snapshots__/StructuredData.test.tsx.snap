// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StructuredData matches snapshot 1`] = `
NodeList [
  <script
    type="application/ld+json"
  >
    {}
  </script>,
]
`;
